### ISSUE TITLE
Reinitialisation of module manager in kiara instance produces a `NotImplemented` exception

### ISSUE THREAD

**Issue by theorm** [DHARPA-Project/kiara#13]
Kiara version `0.0.4`. Can be replicated with this code:

```python
from kiara import Kiara
from kiara.module_mgmt.merged import MergedModuleManager


Kiara.instance()

kiara_instance._module_mgr = MergedModuleManager(
  Kiara.instance()._config.module_managers
)
# raises a NotImplementedError originating from kiara/utils/class_loading.py in find_all_kiara_modules()
```

**Comment by makkus** [comment]
Ok, I won't do any work on this, but just in case this is a simple fix (since the code you posted could never be supported since it randomly changes a private attribute of a class and all the internal state and esp. cache of the object would be invalid). 

What exactly is it you are trying to achieve? Having a new kiara instance, right? Why do you need to re-use the default instance, instead of just creating a custom new one like so:

```
kiara = Kiara()
```
?

**Comment by theorm** [comment]
> What exactly is it you are trying to achieve?

I'm trying to make the new modules installed during the lifetime of the kiara instance available for Kiara.

> Why do you need to re-use the default instance, instead of just creating a custom new one like so:

Because I run into the same issues:

```
>>> from kiara import Kiara
>>> k1 = Kiara()
>>> k2 = Kiara()
# Exception in thread Thread-3:
# ...
# NotImplementedError
```

**Comment by makkus** [comment]
Ok, that makes more sense. Ta.

**Comment by makkus** [comment]
I can't reproduce this, I've included this scenario in my test suite ( https://github.com/DHARPA-Project/kiara/blob/main/tests/test_kiara_context.py ), but it seems to work fine. I even added an extra test and initialized the kiara objects in different threads (since your error message seems to indicate you are using threads), but this also seems to pass without issue.

I'll need more information as to the exact situation where this is happening, so I'll have to ask you to come up with some more code that reproduces your environment more closely, sorry.

One thing to mention (if it isn't obvious): the kiara API is not threadsafe (except for the actual processing of module code). That does not mean kiara can't be used in a threaded environment -- but as is usual in those cases -- you will have to be aware what is happening, and that no state gets modified in other threads (or you sync that state yourself somehow).

**Comment by theorm** [comment]
Fixed in Kiara version `0.1.0`.
